### PR TITLE
Added ENV variable to enable deployment of workflows publically

### PR DESCRIPTION
**Pull request by Josephalexantony-aot**
# Issue Tracking

JIRA: 
Issue Type: BUG/ FEATURE
https://aottech.atlassian.net/browse/FWF-1747
# Changes

Removed check box used for creating wrokflows public, Added the new ENV variable to enable deployment of workflow publically.  Now the user has to change the env value to true for creating new public workflows. 
# How has the change been tested?

# Screenshots (if applicable)

image

# Build Success screenshot (Till a CICD pipeline is set up)

# Notes

### CODE REVIEW

**Review by alanraju-aot — APPROVED** [review]
LGTM

File: forms-flow-web/src/components/Modeller/Editors/BpmnEditor/index.js [hunk 1]
```diff
@@ -79,7 +77,8 @@ export default React.memo(
       }
     }, []);
     useEffect(() => {
-        if (diagramXML) {
+      console.log("dataaaaaaa",tenant, tenantKey, processKey);
```
Comment (sumesh-aot) [inline forms-flow-web/src/components/Modeller/Editors/BpmnEditor/index.js:80]: please remove console log

File: forms-flow-web/src/components/Modeller/Editors/BpmnEditor/index.js [hunk 2]
```diff
@@ -324,12 +319,6 @@ export default React.memo(
         </div>
 
         <div>
-          {MULTITENANCY_ENABLED ? (
```
Comment (sumesh-aot) [inline forms-flow-web/src/components/Modeller/Editors/BpmnEditor/index.js:327]: we would still need this if this feature is enabled.

File: forms-flow-web/src/components/Modeller/Editors/BpmnEditor/index.js [hunk 3]
```diff
@@ -154,7 +148,8 @@ export default React.memo(
       // Deployment Source
       form.append("deployment-source", "Camunda Modeler");
       // Tenant ID
-      if (tenantKey && !applyAllTenants) {
+      //if REACT_APP_PUBLIC_WORKFLOW_ENABLED is false in .env add tenant key to keep it private
+      if (MULTITENANCY_ENABLED && tenantKey && publicWorkflowEnabled === "false" && isNewDiagram) {
```
Comment (sumesh-aot) [inline forms-flow-web/src/components/Modeller/Editors/BpmnEditor/index.js:152]: We wouldn't need this I think, just remove the checkbox if the feature is disabled, then everything should work as expected with current code.

File: forms-flow-web/src/components/Modeller/Editors/DmnEditor/index.js [hunk 4]
```diff
@@ -140,7 +134,7 @@ export default React.memo(
       // Deployment Source
       form.append("deployment-source", "Camunda Modeler");
       // Tenant ID
-      if (tenantKey && !applyAllTenants) {
+      if (MULTITENANCY_ENABLED && tenantKey && publicWorkflowEnabled === "false" && isNewDiagram) {
```
Comment (sumesh-aot) [inline forms-flow-web/src/components/Modeller/Editors/DmnEditor/index.js:137]: Same comment as above

File: forms-flow-web/src/components/Modeller/Editors/BpmnEditor/index.js (hunk 2 above)
Comment (Josephalexantony-aot) [reply forms-flow-web/src/components/Modeller/Editors/BpmnEditor/index.js:327]: now we the user can create the new task as private by default and can create a public one changing the env. so should we need to add this feature?

File: forms-flow-web/src/components/Modeller/Editors/BpmnEditor/index.js (hunk 3 above)
Comment (Josephalexantony-aot) [reply forms-flow-web/src/components/Modeller/Editors/BpmnEditor/index.js:151]: Yes but if we remove this code, then the tenant user can create only the private workflows. once the env is enabled true, then the user can create public workflows too.

File: forms-flow-web/src/components/Modeller/Editors/BpmnEditor/index.js (hunk 1 above)
Comment (Josephalexantony-aot) [reply forms-flow-web/src/components/Modeller/Editors/BpmnEditor/index.js]: done

**Pull request closed without merging**